### ISSUE TITLE
Update WebKit/WebKit Editor detection

### ISSUE THREAD

**Issue by greg-hellings** [crosswire/xiphos#794]
Fedora Rawhide (will be Fedora 27) does not have support for WebKitGTK, and requires WebKit2 APIs. Likewise, both Ubuntu has not had gtkhtml since at least 16.04. Both our usage and detection of the webkit2 package needs to be update to look for the new version (webkit2gtk-4.0.pc instead of webkitgtk-3.0) and the WebKit Editor needs to be finished.

**Comment by karlkleinpaste** [comment]
i could use someone willing to wade into what's wrong with the webkit editor for webkit2. i honestly don't understand why the webkit1 API couldn't have been provided exactly by webkit2.

**Comment by TingPing** [comment]
@karlkleinpaste To be frank you are 7 years late to have that discussion.

**Comment by karlkleinpaste** [comment]
@yetist, i see from insights -> network that you have a webkit2-editor branch. have you done work on this? is there hope you will have something we could merge? you've done such useful work in other gtk matters, i'm hoping you've gotten a handle on this as well.

**Comment by yetist** [comment]
Yes, I created the webkit2-editor branch to support webkit2gtk as the html editor, but the API change of webkit2gtk-4.0 is relatively large,  this work is not very easy, it has not been completed yet. If it is done, I will create PR.

**Comment by jbicha** [comment]
I expect that xiphos will be removed from Debian (unstable) and Ubuntu 19.04 later this year. It has already been removed from Debian Buster ("Testing").

It is impossible to build xiphos on Debian Testing because gtkhtml was removed 3 years ago and the old webkitgtk has finally been removed from Testing too.

**Comment by teusbenschop** [comment]
On Debian (unstable) and therefore on upcoming Ubuntu releases, Xiphos was patched in such a way that it now builds without depending on libwebkitgtk. 

That will keep Xiphos in Debian and Ubuntu.

The patches remove the functionality that libwebkitgtk provided. It's about editing the personal commentary.

The patches can be viewed in the source code repository for the Debian packaging:

https://salsa.debian.org/pkg-crosswire-team/xiphos

**Comment by bgermann** [comment]
I propose making the editor component optional, which would also help with #963. If you agree, I am going to work on that myself.

**Comment by LAfricain** [comment]
It's a good idea. But I'm very interresting about it. And some friends are too! The best way would be to change the editor. However in the meantime it would make compiling and creating the deb package under debian much easier if the editor is optional.

**Comment by greg-hellings** [comment]
@Cloggsy1 Because there are no bugs. Moving to a new version of a package is not a bug, it's an enhancement. That the floor keeps moving out from under us is not an indication that we have a bug. It's only an indication that the floor has moved.

**Comment by karlkleinpaste** [comment]
The more involved answer is that I have had a very difficult time finding motivation to go forward with replacing the current editor with whatever is required by WebKitGTK. It's an annoying segment of Xiphos.

I know that it has been an indecently long time. I will see if I can accomplish something before Christmas. My personal situation has improved and perhaps I can get my mind out of the doldrums.

**Comment by greg-hellings** [comment]
Just poking this thread again, as I received notice this morning that gtkhtml is being orphaned in Fedora. This gives me something like a six week window to either become the Fedora maintainer of it or we lose Xiphos support in Fedora.

Xiphos is the last app in Fedora that links to gtkhtml and, as such, the maintainer of it is dropping maintainership of that library.

Various alternatives are already laid out in the issue about dropping gtk2 code, including all the pros and cons.

**Comment by mfreeman72** [comment]
I personally would really like to have StudyPad back in Xiphos. Any chance that it can be rebased on currently supported libraries, or find a different, but comparable editor that is similar that can be used in its place until this issue is fixed? I have some (admittedly rudimentary) programming experience. Is there anything I might be able to help with on this front?

**Comment by phyrik** [comment]
I take it this issue can also be closed with the recent merges? Obviously we still don't have studypad on various distros but I reckon that warrants a new issue, since this was originally based on detecting if webkit support was present, which has been fixed.

**Comment by karlkleinpaste** [comment]
it's a good point. i'll open a fresh issue to address it directly.
